src/ServiceControl/Hosting/Commands/CustomLivetimes/PersisterInitializingLifetimeWrapper.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using ServiceControl.Persistence;
+
+
+sealed class PersisterInitializingLifetimeWrapper : IHostLifetime
```

Comment:
could we make this more generic i.e. independent of persisted? Could be called `HostLifetimeWithInitializationPhase`

Reply:
It could but then it should also have a constructor that allows for a more generic initialization but I don't think that is needed.